llvm/lib/Target/X86/X86CompressEVEX.cpp

```diff
@@ -154,14 +154,14 @@ static bool performCustomAdjustments(MachineInstr &MI, unsigned NewOpc) {
   case X86::VRNDSCALEPDZ256rmi:
   case X86::VRNDSCALEPSZ256rri:
   case X86::VRNDSCALEPSZ256rmi:
-  case X86::VRNDSCALESDZr:
-  case X86::VRNDSCALESDZm:
-  case X86::VRNDSCALESSZr:
-  case X86::VRNDSCALESSZm:
-  case X86::VRNDSCALESDZr_Int:
-  case X86::VRNDSCALESDZm_Int:
-  case X86::VRNDSCALESSZr_Int:
-  case X86::VRNDSCALESSZm_Int:
+  case X86::VRNDSCALESDZri:
```

Comment:
It's inconsistent with the vector instructions which use `rri`. No idea which one is precise though.

Reply:
Yes, a lot of the scalar variants don't do this consistently, especially when they are unary - so far its something I've been managing to avoid. But rndscale is easier to fix than others (which share more tablegen macrodefs) - I'll see what I can do.